sdk/python/kfp/_client.py

```diff
@@ -88,7 +88,24 @@ def camel_case_to_snake_case(name):
 
 
 class Client(object):
-  """ API Client for KubeFlow Pipeline.
+  """API Client for KubeFlow Pipeline.
+
+  Args:
+      host: The host name to use to talk to Kubeflow Pipelines. If not set, the in-cluster
+        service DNS name will be used, which only works if the current environment is a pod
+        in the same cluster (such as a Jupyter instance spawned by Kubeflow's
+        JupyterHub). If you have a different connection to cluster, such as a kubectl
+        proxy connection, then set it to something like :code:`127.0.0.1:8080/pipeline`.
+        If you connect to an IAP enabled cluster, set it to
+        :code:`https://<your-deployment>.endpoints.<your-project>.cloud.goog/pipeline`.
+      client_id: The client ID used by Identity-Aware Proxy.
+      namespace: The namespace where the kubeflow pipeline system is run.
+      other_client_id: The client ID used to obtain the auth codes and refresh tokens.
+          Reference: https://cloud.google.com/iap/docs/authentication-howto#authenticating_from_a_desktop_app.
+      other_client_secret: The client secret used to obtain the auth codes and refresh tokens.
+      existing_token: Pass in token directly, it's used for cases better get token outside of SDK, e.x. GCP
+          Cloud Functions or caller already has a token
+      cookies: CookieJar object containing cookies that will be passed to the pipelines API.
```

Comment:
Something I noticed consistently was that we were missing the API docs for constructors despite them being there. Looks like Sphinx reads from the class docstring by default, you can have it read from the class docstring or `__init__` or both so happy to conform to either if people have a preference. For now I just kept the Sphinx default.